bin/audit.py

```diff
@@ -0,0 +1,152 @@
+# from gfactoryXmlToDB import *
+from collections import defaultdict
+import yaml
+import requests
+import sys
+import urllib.request
+try:
+    import xml.etree.cElementTree as ET
+except ImportError:
+    import xml.etree.ElementTree as ET
+
+
+treeDump = False
+
+
+def getGfactoryData(gfactoryDB, xml):
+    response = urllib.request.urlopen(xml)
+    gfactoryPage = response.read()
+    root = ET.fromstring(gfactoryPage)
+
+    for entries in root.findall('entries'):
+        for entry in entries.findall('entry'):
+            for attrs in entry.findall('attrs'):
+                for attr in attrs.findall('attr'):
+                    if attr.get('name') == 'GLIDEIN_ResourceName':
+                        if treeDump:
+                            print(attr.get('value'))
+                        # try:
+                        #     gfactoryDB.add(attr.get('value'))
+                        # except KeyError:
+                        #     gfactoryDB = {attr.get('value')}
+                        gfactoryDB.add(attr.get('value'))
+
+
+def getTopologyData(topologyDB):
+    # insert Names under a dictionary that stores 4 "groupname"-set pairs
+    # Structure of the dictionary:
+    # {'resourceGroups': {},
+    #  'facilities': {},
+    #  'sites': {},
+    #  'resources': {}}
+    #
```

Comment:
```suggestion
    """insert Names under a dictionary that stores 4 "groupname"-set pairs
    Structure of the dictionary:
    {'resourceGroups': {},
     'facilities': {},
     'sites': {},
     'resources': {}}
     """

```

This looks like a docstring! https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings